apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -1,191 +1,204 @@
-  });
+const assessmentInstanceRowSchema = z.object({
+  assessment_label: z.string(),
+  user_id: IdSchema.nullable(),
+  uid: z.string().nullable(),
+  name: z.string().nullable(),
+  role: z.string().nullable(),
+  group_id: IdSchema.nullable(),
+  group_name: z.string().nullable(),
+  uid_list: z.array(z.string()).nullable(),
+  user_name_list: z.array(z.string()).nullable(),
+  group_roles: z.array(z.string()).nullable(),
+  username: z.string().nullable(),
+  score_perc: z.number().nullable(),
+  points: z.number(),
+  max_points: z.number(),
+  number: z.number(),
+  assessment_instance_id: IdSchema,
+  open: z.boolean(),
+  time_remaining: z.string(),
+  time_remaining_sec: z.number().nullable(),
+  total_time: z.string(),
+  total_time_sec: z.number().nullable(),
+  date: z.date(),
+  date_formatted: z.string(),
+  duration: z.string(),
+  duration_secs: z.number(),
+  duration_mins: z.number(),
+  highest_score: z.boolean(),
+  client_fingerprint_id_change_count: z.number(),
 });
 
-router.get('/client.js', function (req, res, next) {
-  debug('GET /client.js');
-  if (!res.locals.authz_data.has_course_instance_permission_view) {
-    return next(error.make(403, 'Access denied (must be a student data viewer)'));
-  }
-  res.type('text/javascript');
-  res.render(__filename.replace(/\.js$/, 'ClientJS.ejs'), res.locals);
-});
+router.get(
+  '/raw_data.json',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    const result = await sqldb.queryRows(
+      sql.select_assessment_instances,
+      {
+        assessment_id: res.locals.assessment.id,
+        group_work: res.locals.assessment.group_work,
+      },
+      assessmentInstanceRowSchema,
+    );
+    res.send(result);
+    return;
+  }),
+);
 
-router.get('/', function (req, res, next) {
-  debug('GET /');
-  if (!res.locals.authz_data.has_course_instance_permission_view) {
-    return next(error.make(403, 'Access denied (must be a student data viewer)'));
-  }
-  debug('render page');
-  res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-});
+router.get(
+  '/client.js',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    res.type('text/javascript');
+    res.render(__filename.replace(/\.js$/, 'ClientJS.ejs'), res.locals);
+  }),
+);
 
-router.post('/', function (req, res, next) {
-  if (!res.locals.authz_data.has_course_instance_permission_edit) {
-    return next(error.make(403, 'Access denied (must be a student data editor)'));
-  }
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_view) {
+      throw error.make(403, 'Access denied (must be a student data viewer)');
+    }
+    debug('render page');
```

Comment:
I don't think this is adding much; consider removing it as long as you're touching this code.

Reply:
This has now been removed in commit [8472ae8](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/8472ae8a0869182f87c71e15b748cbd6064db927).